### ISSUE TITLE
Assertion fail in jpc_math.c:89

### ISSUE THREAD

**Issue by seviezhou** [jasper-software/jasper#225]
## System info

Ubuntu X64, gcc (Ubuntu 5.5.0-12ubuntu1), jasper (latest master [80fd24](https://github.com/jasper-software/jasper/commit/80fd248c50bab618436fa0d1474c5b6d32ae9332))

## Configure

CFLAGS="-g -fsanitize=address" LDFLAGS="-fsanitize=address" ./configure

## Command line

./jasper --input @@ --output /dev/null --output-format pnm

## Output

```
jasper: /home/seviezhou/jasper/src/libjasper/jpc/jpc_math.c:89: int jpc_floorlog2(int): Assertion `x > 0' failed.
Aborted (core dumped)
```

## POC

[assert-fail-jpc_math-89.zip](https://github.com/jasper-software/jasper/files/5012212/assert-fail-jpc_math-89.zip)

**Comment by osamu620** [comment]
I have check the attached codestream. This codestream has some violations of the codestream syntax defined in the spec. I think this is not a bug of JasPer. Of course JasPer shall check the codestream syntax and say "this input cannot be decodable."

**Comment by osamu620** [comment]
I have analyzed what JasPer did to the attached codestream. Although this codestream looks very weird, it has the Scod parameter = 0x05. The 3rd LSB of the Scod means EPH marker was used(1) or not(0). So, the Scod parameter said this codestream used EPH. Then decoders shall check the existence of EPH marker(0xFF92) at the end of the packet header parsing.
The following part is the code lines for the checking.
 
https://github.com/jasper-software/jasper/blob/80fd248c50bab618436fa0d1474c5b6d32ae9332/src/libjasper/jpc/jpc_t2dec.c#L367-L380

I think the lines 368 and 379 shall be deleted because the `jpc_dec_lookahead()` which is called in the line 368 assumes successive two bytes in the bitstream would never be out of the range from 0xFF80 to 0xFFFF. Of course, this assumption is wrong. I have checked this modification works pretty well and never break the conformance tests.
